CRM/Attentively/BAO/Attentively.php

```diff
@@ -0,0 +1,275 @@
+          VALUES ( '{$value->member_id}', '{$value->contact_id}', '{$value->email_address}', '{$value->first_name}', '{$value->last_name}', '{$value->age}', '{$value->city}', '{$value->state}', 
+          '{$value->zip_code}', '{$value->metro_area}', '{$value->klout_score}' ) 
+          ON DUPLICATE KEY UPDATE member_id = '{$value->member_id}', email_address = '{$value->email_address}', first_name = '{$value->first_name}', last_name = '{$value->last_name}',
+          age = '{$value->age}', city = '{$value->city}', state = '{$value->state}', zip_code = '{$value->zip_code}', metro_area = '{$value->metro_area}', klout_score = '{$value->klout_score}'";
+        $dao = CRM_Core_DAO::executeQuery($sql);
+        // Store networks
+        foreach ($value->networks as $key => $networks) {
+          $network[$key]['contact_id'] = $value->contact_id;
+          $network[$key]['name'] = $networks->name;
+          $network[$key]['url'] = $networks->url;
+        }
+      }
+      foreach ($network as $key => $value) {
+        $query = "INSERT INTO civicrm_attentively_member_network (`contact_id`, `name`, `url`)
+          VALUES ( '{$value['contact_id']}', '{$value['name']}', '{$value['url']}' )";
+        $dao = CRM_Core_DAO::executeQuery($query);
+      }
+    }
+  }
+
+  static public function pullPosts($terms) {
+    if (empty($terms)) {
+      return;
+    }
+    $settings = CRM_Core_OptionGroup::values('attentively_auth', TRUE, FALSE, FALSE, NULL, 'name', FALSE);
+    $url = self::checkEnvironment();
+    $url = $url . 'posts';
+    $post = 'access_token=' . $settings['access_token'] . '&period=' . $settings['post_period_to_retrieve'] . '&term=' . $terms . '&include_member_info=TRUE';
+    $ch = curl_init( $url );
+    curl_setopt( $ch, CURLOPT_POST, TRUE);
+    curl_setopt( $ch, CURLOPT_POSTFIELDS, $post);
+    curl_setopt( $ch, CURLOPT_FOLLOWLOCATION, 1);
+    curl_setopt( $ch, CURLOPT_HEADER, 0);
+    curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);
+    $response = curl_exec( $ch );
+    $result = get_object_vars(json_decode($response));
+    return $result;
+  }
+
+  static public function pushWatchedTerms($terms) {
+    if (empty($terms)) {
+      return;
+    }
+    $settings = CRM_Core_OptionGroup::values('attentively_auth', TRUE, FALSE, FALSE, NULL, 'name', FALSE);
+    $url = self::checkEnvironment();
+    $url = $url . 'watched_terms_add';
+    $post = 'access_token=' . $settings['access_token'] . '&terms=' . $terms;
+    $ch = curl_init( $url );
+    curl_setopt( $ch, CURLOPT_POST, TRUE);
+    curl_setopt( $ch, CURLOPT_POSTFIELDS, $post);
+    curl_setopt( $ch, CURLOPT_FOLLOWLOCATION, 1);
+    curl_setopt( $ch, CURLOPT_HEADER, 0);
+    curl_setopt( $ch, CURLOPT_RETURNTRANSFER, 1);
+    $response = curl_exec( $ch );
+    $result = get_object_vars(json_decode($response));
+    return $result;
+  }
+
```

Comment:
Can't we use as a sinlge function for pushWatchedTerms(), pullPosts(), pullMember(). I see many code duplicates.